lib/datasource/index.ts

```diff
@@ -298,6 +298,36 @@ export async function getPkgReleases(
         (findRelease) => findRelease.version === filterRelease.version
       ) === filterIndex
   );
+  // Filter releases for compatibility
+  for (const [constraintName, constraintValue] of Object.entries(
+    config.constraints || {}
+  )) {
+    if (version.isVersion(constraintValue)) {
+      res.releases = res.releases.filter((release) => {
+        if (!release.constraints) {
+          return true;
+        }
+        if (!release.constraints[constraintName]) {
+          return true;
+        }
```

Comment:
i think this can be d
```suggestion
        if (!release.constraints?.[constraintName]) {
          return true;
        }
```